src/OpenSpaceToolkit/Astrodynamics/Trajectory/Segment.cpp

```diff
@@ -11,20 +14,124 @@ namespace trajectory
 
 using ostk::astro::trajectory::Propagator;
+using ostk::astro::trajectory::state::CoordinatesSubset;
 
 Segment::Solution::Solution(
     const String& aName,
     const Array<Shared<Dynamics>>& aDynamicsArray,
     const Array<State>& aStates,
-    const bool& aConditionIsSatisfied
+    const bool& aConditionIsSatisfied,
+    const Segment::Type& aSegmentType
 )
     : name(aName),
       dynamics(aDynamicsArray),
       states(aStates),
-      conditionIsSatisfied(aConditionIsSatisfied)
+      conditionIsSatisfied(aConditionIsSatisfied),
+      segmentType(aSegmentType)
+{
+}
+
+const Instant& Segment::Solution::accessStartInstant() const
+{
+    if (states.isEmpty())
+    {
+        throw ostk::core::error::RuntimeError("No solution available.");
+    }
+
+    return states.accessFirst().accessInstant();
+}
+
+const Instant& Segment::Solution::accessEndInstant() const
+{
+    if (states.isEmpty())
+    {
+        throw ostk::core::error::RuntimeError("No solution available.");
+    }
+
+    return states.accessLast().accessInstant();
+}
+
+Mass Segment::Solution::getInitialMass() const
 {
+    if (states.isEmpty())
+    {
+        throw ostk::core::error::RuntimeError("No solution available.");
+    }
+
+    return Mass::Kilograms(states.accessFirst().extractCoordinates(CoordinatesSubset::Mass())[0]);
+}
+
+Mass Segment::Solution::getFinalMass() const
+{
+    if (states.isEmpty())
+    {
+        throw ostk::core::error::RuntimeError("No solution available.");
+    }
+
+    return Mass::Kilograms(states.accessLast().extractCoordinates(CoordinatesSubset::Mass())[0]);
+}
+
+Real Segment::Solution::computeDeltaV(const Real& aSpecificImpulse) const
+{
+    // TBM: This is only valid for constant thrust, constant Isp
+    if (segmentType != Segment::Type::Maneuver)
+    {
+        return 0.0;
+    }
+
+    return aSpecificImpulse * EarthGravitationalModel::gravityConstant *
+           std::log(getInitialMass().inKilograms() / getFinalMass().inKilograms());
+}
+
+Mass Segment::Solution::computeDeltaMass() const
+{
+    if (segmentType != Segment::Type::Maneuver)
+    {
+        return Mass::Kilograms(0.0);
+    }
+
+    return Mass::Kilograms(getInitialMass().inKilograms() - getFinalMass().inKilograms());
+}
+
+void Segment::Solution::print(std::ostream& anOutputStream, bool displayDecorator) const
+{
+    if (displayDecorator)
+    {
+        ostk::core::utils::Print::Header(anOutputStream, "Segment Solution");
+    }
+
+    ostk::core::utils::Print::Line(anOutputStream) << "Name:" << name;
+    ostk::core::utils::Print::Line(anOutputStream)
+        << "Condition satisfied:" << (conditionIsSatisfied ? "True" : "False");
+    ostk::core::utils::Print::Line(anOutputStream)
+        << "Segment type:" << (segmentType == Segment::Type::Coast ? "Coast" : "Maneuver");
+
+    ostk::core::utils::Print::Line(anOutputStream) << "Start instant:" << accessStartInstant().toString();
```

Comment:
Should we check here wether `states.isEmpty()` before printing some of the fields below?

Just to avoid printing:

```
[...]
Start instant: No solution available.
Final instant: No solution available.
Initial mass: No solution available.
[etc.]
```

Reply:
It would raise an error if that's the case. However, we shouldn't really have empty sequencesolutions, there should always be a segment even if it aborted early. So I think we're covered.